migration-tools/migrate-rosdistro.py

```diff
@@ -221,8 +221,19 @@ def write_tracks_file(tracks, commit_msg=None):
         # Bloom will not run with multiple remotes.
         subprocess.check_call(['git', 'remote', 'remove', 'oldorigin'])
         subprocess.check_call(['git', 'bloom-release', '--non-interactive', '--release-increment', release_inc, '--unsafe', args.dest], stdin=subprocess.DEVNULL, env=os.environ)
-        subprocess.check_call(['git', 'push', 'origin', '--all', '--force'])
-        subprocess.check_call(['git', 'push', 'origin', '--tags', '--force'])
+
+        # If something goes wrong, use dry run pushes to ensure we don't
+        # destroy any data in the repository.
+        # 1. Ensure we're not overwriting tags
+        # 2. Push `upstream` branch. This is the ONLY branch we should ever
+        #    force-push, and is an intermediate staging branch anyway.
+        # 3. Ensure we're not overwriting any branches
+        # 4. Push the branches and tags
+        subprocess.check_call(['git', 'push', 'origin', '--tags', '--dry-run'])
+        subprocess.check_call(['git', 'push', 'origin', 'upstream', '--force'])
+        subprocess.check_call(['git', 'push', 'origin', '--all', '--dry-run'])
+        subprocess.check_call(['git', 'push', 'origin', '--tags', '--all'])
```

Comment:
Putting my observation into suggestion.
```suggestion
        subprocess.check_call(['git', 'push', 'origin', '--tags'])
        subprocess.check_call(['git', 'push', 'origin', '--branches'])
```